test/Models/TerriaSpec.ts

```diff
@@ -332,6 +374,57 @@ describe("Terria", function() {
   });
 
   describe("updateApplicationUrl", function() {
+    it("works with initializationUrls and initFragmentPaths", async function() {
+      expect(terria.initSources.length).toEqual(0);
+
+      jasmine.Ajax.install();
+      // Fail all requests by default.
+      jasmine.Ajax.stubRequest(/.*/).andError({});
+
+      jasmine.Ajax.stubRequest("path/to/config/configUrl.json").andReturn({
+        responseText: JSON.stringify({
+          initializationUrls: ["something"],
+          parameters: {
+            applicationUrl: "https://application.url/",
+            initFragmentPaths: [
+              "path/to/init/",
+              "https://hostname.com/some/other/path/"
+            ]
+          }
+        })
+      });
+
+      await terria.start({
+        configUrl: `path/to/config/configUrl.json`,
+        i18nOptions
+      });
+
+      await terria.updateApplicationUrl(
+        "https://application.url/#someInitHash"
+      );
+
+      console.log(terria.initSources);
+
+      expect(terria.initSources.length).toEqual(2);
+
+      const initSource = terria.initSources[1];
+      expect(isInitFromOptions(initSource)).toBeTruthy();
+
+      if (!isInitFromOptions(initSource))
+        throw "Init source is not from options";
+
+      expect(
+        initSource.options.map(source =>
+          isInitFromUrl(source) ? source.initUrl : ""
+        )
+      ).toEqual([
+        "https://application.url/path/to/init/someInitHash.json",
```

Comment:
Need to explain why it is not "https://application.url/path/to/config/path/to/init/someInitHash.json" when initFragmentPaths[0] is set to "path/to/init/", which has different result from line 179.

It seems that the path specified by configUrl will be ignored.

Reply:
I have added comments to how paths are constructed

I also added more docs

- https://github.com/TerriaJS/terriajs/blob/31b8384535768d958b2d653486bcaed5b7a7dc85/doc/deploying/controlling-with-url-parameters.md#catalog-files-init-files
- https://github.com/TerriaJS/terriajs/blob/31b8384535768d958b2d653486bcaed5b7a7dc85/doc/customizing/client-side-config.md#intializationurls